### PR TITLE
Additional license fixes

### PR DESCRIPTION
**Pull request by dthaler**
The Apache-2.0 and NASA-1.3 licenses require inclusion of licenses
in this project (much like the [Crab](https://github.com/seahorn/crab) project [includes](https://github.com/seahorn/crab/blob/master/IKOS_LICENSE.pdf) the NASA-1.3
license in addition to the Apache-2.0 license file).

Also, per the Apache 2.0 license terms, files that are directly derived
from the Crab project under Apache 2.0 need to stay under Apache-2.0 so
this PR corrects the SPDX labels on those files to be Apache-2.0 instead
of MIT.  This affects many, but not all, files under src/crab and src/crab_utils.

Finally, this PR adds SPDX ids to the source files under the external directory.

Addresses more of issue #104 

Signed-off-by: Dave Thaler <dthaler@ntdev.microsoft.com>

### CODE REVIEW

**Comment by dthaler** [comment]
@caballa Please check

File: scripts/.check-license.ignore [hunk 1]
```diff
@@ -4,23 +4,50 @@
...
 # Directories using GPL
 src/gpl
 counter
 
 # Other Directories
 ebpf-samples
-external
 
 # Files using NASA-1.3 license
+src/crab/array_domain.cpp
```
Comment (caballa) [inline scripts/.check-license.ignore:20]: It should go with Apache 2.0

Comment (dthaler) [reply scripts/.check-license.ignore]: Fixed, thanks

**Comment by elazarg** [comment]
LGTM. @caballa feel free to merge.

**Comment by caballa** [comment]
I don't have the option to merge PRs

**Comment by elazarg** [comment]
Now you do :)

**Pull request merged**